### ISSUE TITLE
Ignore classes for all rules

### ISSUE THREAD

**Issue by lsrom** [detekt/detekt#1340]
Hi, I noticed only some rules (LateinitUsage) have the ability to ignore classes based on regex. I would like to add this functionality to other rules as well. In my use case I need it for UnsafeCallOnNullableType but I think it would be beneficial to add this to all rules. 

My idea is to add it to BaseRule class which would auto-magically add it to all rules. Is this OK acceptable change? Are there any drawbacks I'm not aware? I only glanced over the code so maybe I'm missing something but if not, I would be happy to send a MR :)

**Comment by arturbosch** [comment]
It would be a good idea actually. Like we have added alias feature for all rules.
Need to look if this is actually possible with current architecture.

**Comment by lsrom** [comment]
Hi, did you have any chance of checking it out? Can I send merge request?

**Comment by arturbosch** [comment]
Hey, if you come up with a solution feel free to open a PR and we can discuss it, thanks!

**Comment by sowmyav24** [comment]
There is  a `Rule` class which has the method `shouldRunOnGivenFile`. Would it make sense to check for base-class configuration here as all the other rules extend from `Rule`

**Comment by schalkms** [comment]
@sowmyav24 yes it would make sense.
@lsrom do you have a PR with a working solution?

**Comment by sowmyav24** [comment]
How would  the configuration for this look like ? Would it make sense to have a property `IgnoreClass` with a list of `IGNORE_ON_CLASSES_PATTERN` that would be checked for all the rules configured. Or would it make sense to have this property at RuleSet level, For eg: `comments`, `complexity` etc

**Comment by BraisGabin** [comment]
Can't this be archived with exclude?

**Comment by sowmyav24** [comment]
Exclude is present at individual rules if I am not wrong. My understanding of the issue was, having a global exclude.

**Comment by BraisGabin** [comment]
The detekt gradle plugin has options to include/exclude files so they are not part of the analysis so maybe that's a way to accomplish that. A bit offtopic but maybe we have problems with this. Because right now the list of files to review and the list of files that we use to generate our `BindingContext` is the same... and probably they should not be the same.

**Comment by BraisGabin** [comment]
With the introduction of suppressors it would be easy to implement this. But why would we need to avoid based on the name of a class? Wouldn't it have more sense to exclude based on its hierarchy? I mean. Instead of exclude the classes with name `.*Activity` we could ignore all the classes that extend from `android.activity`. Our all the classes that implements `OnClickListener`

**Comment by schalkms** [comment]
@BraisGabin I think this is a different use case but also a valid one of course.

**Comment by BraisGabin** [comment]
But when do you want to filter just based on the name of the class? Isn't any other thing that identifies better those classes? Extends, implements, an annotation...

**Comment by BraisGabin** [comment]
I just read the rationale for `LateinitUsage.ignoreOnClassesPattern` in #207 and I think that it doesn't apply right now. I think that we should even deprecate it.

And quoting myself here:

> Can't this be archived with exclude?

```yml
AnyRule:
  exclues: ["**/*Activity.kt]]
```

For this reason I'm closing this. Please, don't hestiate to write in this issue (or open a new one) if I'm missing something here.
